### PR TITLE
casper-js-sdk 2.8.0

### PR DESCRIPTION
**Pull request by hoffmannjan**
- added support to `CLPublicKey` passed as argument to `CLKey` (it gets serialized to `account-hash` when serializing).
- changed `CLKey` method checking stored value type from `isAccount` to `isAccountHash`.
- new `Contracts` class that basically acts as contract client (possible to install WASM, call an entrypoint, query state or dictionary).
- added `getDictionaryItemByName` that uses `state_get_dictionary_item` with `dictionary_name` and `dictionary_item_key`.
- added `sign` and `send` method on `Deploy` class.

### CODE REVIEW

File: src/lib/Contracts.ts [hunk 1]
```diff
@@ -19,112 +17,174 @@ export function byteHash(x: Uint8Array): Uint8Array {
...
-    const paymentArgs = RuntimeArgs.fromMap({
-      amount: CLValueBuilder.u512(paymentAmount.toString())
-    });
-
-    const payment = ExecutableDeployItem.newModuleBytes(
-      this.paymentWasm,
-      paymentArgs
+    sender: CLPublicKey,
+    chainName: string,
+    paymentAmount: string,
```
Comment (HansWestman74) [inline src/lib/Contracts.ts:78]: Ideally, the developer facing function should not be able to provide a string that cannot be convert to a bigint.

File: src/lib/Contracts.ts [hunk 2]
```diff
@@ -19,112 +17,174 @@ export function byteHash(x: Uint8Array): Uint8Array {
...
+    signingKeys: AsymmetricKey[] = [],
+    ttl: number = DEFAULT_DEPLOY_TTL
+  ): Deploy {
+    this.checkSetup();
+
+    const contractHashAsByteArray = contractHashToByteArray(
+      this.contractHash!.slice(5)
```
Comment (HansWestman74) [inline src/lib/Contracts.ts:85]: Add a check that it begins with "hash-"

File: src/lib/Contracts.ts [hunk 3]
```diff
@@ -19,112 +17,174 @@ export function byteHash(x: Uint8Array): Uint8Array {
...
-   * Arguments for Faucet smart contract
-   *
-   * @param accountPublicKeyHash the public key hash that want to be funded
-   */
-  public static args(accountPublicKeyHash: Uint8Array): RuntimeArgs {
-    const accountKey = new CLKey(new CLAccountHash(accountPublicKeyHash));
-    return RuntimeArgs.fromMap({
-      account: accountKey
-    });
+  public async queryContractState(
```
Comment (HansWestman74) [inline src/lib/Contracts.ts]: Should delete this

**Review by HansWestman74 — APPROVED** [review]

File: src/lib/Contracts.ts (hunk 1 above)
Comment (hoffmannjan) [reply src/lib/Contracts.ts:78]: I just checked and when I provided "ABC" as a payment amount this error got thrown
`Error: invalid BigNumber string (argument="value", value="ABC", code=INVALID_ARGUMENT, version=bignumber/5.1.1)`
from my perspective this is currently enough, there are many places where we want user to provide numeric as a string and adding checks like this at every function would be overkill IMO.

File: src/lib/Contracts.ts (hunk 3 above)
Comment (hoffmannjan) [reply src/lib/Contracts.ts]: done :)

File: src/lib/Contracts.ts (hunk 2 above)
Comment (hoffmannjan) [reply src/lib/Contracts.ts:85]: done, added check in `setContractHash` method :)

**Review by George-cl — APPROVED** [review]
LGTM - do you need the `src/types.ts` file if it's empty?

**Comment by hoffmannjan** [comment]
> LGTM - do you need the `src/types.ts` file if it's empty?

nope, removing it, thx :)

**Pull request merged**